Check ARM Neoverse build with `-DENABLE_WARNINGS=ON`

The following error shows up on Juno R2 with `-DENABLE_WARNINGS=ON`, but unclear if we need the `socketid` or `cpuid` in the future. To be cross-checked with @amarathe84. 

```
/mnt/ssd/shared-patki1/variorum/src/variorum/ARM/neoverse_N1_power_features.c:275:50: error: unused parameter 'socketid' [-Werror=unused-parameter]
 int arm_cpu_neoverse_n1_cap_socket_frequency(int socketid, int new_freq)
                                                  ^
cc1: all warnings being treated as errors
make[2]: *** [variorum/ARM/CMakeFiles/variorum_arm.dir/build.make:135: variorum/ARM/CMakeFiles/variorum_arm.dir/neoverse_N1_power_features.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:161: variorum/ARM/CMakeFiles/variorum_arm.dir/all] Error 2
make: *** [Makefile:128: all] Error 2
 ```

--- a/src/variorum/ARM/neoverse_N1_power_features.c
+++ b/src/variorum/ARM/neoverse_N1_power_features.c
@@ -206,7 +206,7 @@ int arm_cpu_neoverse_n1_get_clocks_data(int chipid, int verbose, FILE *output)
 
     for (core_iter = 0; core_iter < NUM_CORES; core_iter++)
     {
-        sprintf(freq_fname, "%s%d/scaling_cur_freq", freq_path, core_iter);
+        sprintf(freq_fname, "%s%lu/scaling_cur_freq", freq_path, core_iter);
         int freq_fd = open(freq_fname, O_RDONLY);
         if (!freq_fd)
         {
@@ -274,16 +274,13 @@ int arm_cpu_neoverse_n1_get_clocks_data(int chipid, int verbose, FILE *output)
 
 int arm_cpu_neoverse_n1_cap_socket_frequency(int socketid, int new_freq)
 {
-    static int init_output = 0;
     uint64_t core_iter;
-    uint64_t aggregate_freq = 0;
-
     char freq_fname[4096];
     char *freq_path = "/sys/devices/system/cpu/cpufreq/policy";
 
     for (core_iter = 0; core_iter < NUM_CORES; core_iter++)
     {
-        sprintf(freq_fname, "%s%d/scaling_setspeed", freq_path, core_iter);
+        sprintf(freq_fname, "%s%lu/scaling_setspeed", freq_path, core_iter);
         int freq_fd = open(freq_fname, O_WRONLY);
         if (!freq_fd)
         {

